src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -169,6 +171,45 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal {
 
 	}
 
+	async getLinks(linkManager: TerminalLinkManager): Promise<IDetectedLinks> {
```

Comment:
This actually introduces a circular dependency:

```
XtermTerminal -> TerminalLinkManager -> XtermTerminal
```

This will probably end up causing us problems because of how large and complex `TerminalLinkManager` is.

How about exposing getLinks as a function in terminalInstance.ts instead and moving the tests to terminalInstance.test.ts to resolve this?

```ts
export function getLinks(terminal: Terminal, linkManager: TerminalLinkManager): Promise<IDetectedLinks>
```

Reply:
it looked like a ton of test infrastructure would be involved with adding this test to `terminalInstance.test.ts`. maybe I should just revert this?

Reply:
what about moving this all into link manager, since the link manager has an `xtermTerminal`?